pyaedt/icepak.py

```diff
@@ -6047,3 +5914,192 @@ def assign_conducting_plate_with_conductance(self, obj_plate, boundary_name=None
                                             shell_conduction=shell_conduction,
                                             low_side_rad_material=low_side_rad_material,
                                             high_side_rad_material=high_side_rad_material)
+
+    @pyaedt_function_handler
+    def _create_dataset_assignment(self, type_assignment, ds_name, scale):
+        ds = None
+        try:
+            if ds_name.startswith("$"):
+                raise AttributeError("Only design datasets are supported.")
+            else:
+                ds = self.design_datasets[ds_name]
+        except KeyError:
+            raise AttributeError("Dataset {} not found.".format({ds_name}))
+        if not isinstance(scale, str):
+            scale = str(scale)
+        return PieceWiseLinearDictionary(type_assignment, ds, scale)
```

Comment:
```suggestion
    @pyaedt_function_handler
    def __create_dataset_assignment(self, type_assignment, ds_name, scale):
        """Create dataset condition assignments.
        
        Parameters
        ----------
        type_assignment : str
            Type of assignment represented by the class.
            Options are ``"Temp Dep"`` and ``"Transient"``.
        ds_name : str
            Dataset name to assign.
        scale : str
            Scaling factor for the y values of the dataset.

        Returns
        -------
        bool or :class:`PieceWiseLinearDictionary`
            Created dataset condition assignments when successful, ``False`` when failed.
        """
        ds = None
        try:
            if ds_name.startswith("$"):
                self.logger.error("Only design datasets are supported.")
                return False
            else:
                ds = self.design_datasets[ds_name]
        except KeyError:
            self.logger.error("Dataset {} not found.".format({ds_name}))
            return False
        if not isinstance(scale, str):
            scale = str(scale)
        return PieceWiseLinearDictionary(type_assignment, ds, scale)
```
The current way of doing things in pyaedt is to return False when something goes wrong.
Note that I think we should raise Exception instead of returning False but I think that this should be done in some refactoring of the whole code base. IMO it would be better to be consistent here.

Also, I've the feeling that this method should not be directly accessible and made private so I changed the "visibility".